### PR TITLE
PR to make purge script more robust for MacOS

### LINKED ISSUE #16404: issues with nix purge script
# Bug Report

## Problem 

The purge script `make nix-purge` often fails with 

```log
Are you sure you want to purge Nix?
[y/n]: y
Detected multi-user Nix installation.
/Library/LaunchDaemons/org.nixos.darwin-store.plist: Could not find specified service
Unload failed: 113: Could not find specified service
make: *** [nix-purge] Error 3
```

and other times it fails with 

```log
Detected multi-user Nix installation.
Started APFS operation
Deleting APFS Volume from its APFS Container
Unmounting disk3s7
The volume "Nix Store" on disk3s7 couldn't be unmounted because it is in use by process 0 (kernel)
Error: -69888: Couldn't unmount disk
```

as reported by @flexsurfer on macOS rosetta terminal. 

It would be better if the make command for nix purge can handle these errors and allow us to re-run the purge command without doing `/nix/scripts/purge.sh --force`...

### PR DESCRIPTION
**Pull request by siddarthkay**
The existing purge script for Darwin would fail and stop at many stages.

This PR makes the script more robust to certain failures that were observed and documented in https://github.com/status-im/status-mobile/issues/16404

fixes #16404

status: ready

### CODE REVIEW

**Review by BalogunofAfrica — APPROVED** [review]
LGTM

**Comment by siddarthkay** [comment]
@status-im/mobile-qa : merged without QA because this change only effects local dev environments.

File: nix/scripts/purge.sh [hunk 1]
```diff
@@ -41,15 +41,39 @@ nix_purge_darwin_multi_user_users() {
...
+
+            # Ask the user whether to terminate the process
+            read -p "Do you want to terminate this process? (y/n): " choice
+            if [[ $choice == "y" ]]; then
+                sudo kill $pid
+                echo "Process $pid terminated."
+            else
+                echo "Process not terminated. Please close it manually and retry."
+                return 1
+            fi
```
Comment (jakubgs) [inline nix/scripts/purge.sh:62]: This whole thing should be a function. Or even a separate `scripts/kill_proc_prompt.sh`.

File: nix/scripts/purge.sh [hunk 2]
```diff
@@ -25,8 +25,8 @@ nix_purge_darwin_multi_user_service() {
     cd /Library/LaunchDaemons
     NIX_SERVICES=(org.nixos.darwin-store.plist org.nixos.nix-daemon.plist)
     for NIX_SERVICE in "${NIX_SERVICES[@]}"; do
-        sudo launchctl unload "${NIX_SERVICE}"
-        sudo launchctl remove "${NIX_SERVICE}"
+        sudo launchctl unload "${NIX_SERVICE}" || true
+        sudo launchctl remove "${NIX_SERVICE}" || true
```
Comment (jakubgs) [inline nix/scripts/purge.sh:29]: This isn't "robust", this is just ignoring failures.

File: nix/scripts/purge.sh [hunk 3]
```diff
@@ -41,15 +41,39 @@ nix_purge_darwin_multi_user_users() {
...
 }
 
 nix_purge_multi_user() {
     if [[ $(uname -s) == "Darwin" ]]; then
-        nix_purge_darwin_multi_user_service
-        nix_purge_darwin_multi_user_users
-        nix_purge_darwin_multi_user_volumes
+        nix_purge_darwin_multi_user_service || true
+        nix_purge_darwin_multi_user_users || true
+        nix_purge_darwin_multi_user_volumes || true
```
Comment (jakubgs) [inline nix/scripts/purge.sh:76]: Same here, this is not "robust", it just ignores failures and continues.

File: nix/scripts/purge.sh (hunk 2 above)
Comment (jakubgs) [reply nix/scripts/purge.sh:29]: We need to understand why `unload` would fail.

**Comment by jakubgs** [comment]
I do not like any of the `|| true` stuff you added, it needs to be fixed.

**Comment by siddarthkay** [comment]
@jakubgs : I agree with your review on this PR, the intent back then was to make sure that the purge script would run without exiting when it encountered any failures and so that the devs  would be able to purge nix via this script without the need for any manual intervention.

I created https://github.com/status-im/status-mobile/issues/18360 to detail out the issues with the existing steps so that we can perform a proper fix.

Thank you

**Pull request merged**